### ISSUE TITLE
Infinite formatting with a multi-line interpolated string

### ISSUE THREAD

**Issue by MaxDesiatov** [nicklockwood/SwiftFormat#613]
I have this snippet:

```swift
import HypertextLiteral

func layout(root: String, body: HTML) -> HTML {
  #"""
    <html>
      <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          \#([16, 32, 96].map { side -> HTML in
            let size = "\(side)x\(side)"
            return #"""
            <link
              rel="icon"
              type="image/png"
              sizes="\#(size)"
              href="\#(root)/static/imgs/favicon-\#(size).png" />
            """#
            }
          )
      </head>
    </html>
  """#
}
```

With this config:

```
--indent 2
--indentcase false
--trimwhitespace always
--ranges nospace
--empty tuple
--operatorfunc nospace
--ifdef noindent
--stripunusedargs closure-only
--disable andOperator
--swiftversion 5.2
```

After formatting, the indentation within the interpolated bit is increased, but after another formatting run it is increased even further, here's how it looks after the first run:

```swift
  #"""
    <html>
      <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          \#([16, 32, 96].map { side -> HTML in
              let size = "\(side)x\(side)"
              return #"""
              <link
                rel="icon"
                type="image/png"
                sizes="\#(size)"
                href="\#(root)/static/imgs/favicon-\#(size).png" />
              """#
              
  }
          )
      </head>
    </html>
  """#
```

and a second one:

```swift
  #"""
    <html>
      <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          \#([16, 32, 96].map { side -> HTML in
                let size = "\(side)x\(side)"
                return #"""
                <link
                  rel="icon"
                  type="image/png"
                  sizes="\#(size)"
                  href="\#(root)/static/imgs/favicon-\#(size).png" />
                """#
                
    
  }
          )
      </head>
    </html>
  """#
```

**Comment by nicklockwood** [comment]
Thanks, I'll investigate.

**Comment by MaxDesiatov** [comment]
I've been able to narrow it down to the multiline interpolation bits (not counting for the inner multi-line string), when the interpolation is on a single line it's fine:

```swift
  #"""
    <html>
      <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          \#([16, 32, 96].map { side -> HTML in let size = "\(side)x\(side)"; return #"""
            <link
              rel="icon"
              type="image/png"
              sizes="16x16"
              href="\#(root)/static/imgs/favicon-\#(size).png" />
            """# })
      </head>
    </html>
  """#
```

**Comment by nicklockwood** [comment]
Fixed in 0.44.8

**Comment by MaxDesiatov** [comment]
Fantastic, thank you so much @nicklockwood!
